command/job_init.go

```diff
@@ -90,27 +101,87 @@ func (c *JobInitCommand) Run(args []string) int {
 		c.Ui.Error(fmt.Sprintf("Failed to stat '%s': %v", filename, err))
 		return 1
 	}
-	if !os.IsNotExist(err) {
+	if !os.IsNotExist(err) && !listTemplates {
 		c.Ui.Error(fmt.Sprintf("Job '%s' already exists", filename))
 		return 1
 	}
 
 	var jobSpec []byte
-	switch {
-	case connect && !short:
-		jobSpec, err = Asset("command/assets/connect.nomad")
-	case connect && short:
-		jobSpec, err = Asset("command/assets/connect-short.nomad")
-	case !connect && short:
-		jobSpec, err = Asset("command/assets/example-short.nomad")
-	default:
-		jobSpec, err = Asset("command/assets/example.nomad")
-	}
-	if err != nil {
-		// should never see this because we've precompiled the assets
-		// as part of `make generate-examples`
-		c.Ui.Error(fmt.Sprintf("Accessed non-existent asset: %s", err))
-		return 1
+
+	if listTemplates {
+		// Get the HTTP client
+		client, err := c.Meta.Client()
+		if err != nil {
+			c.Ui.Error(fmt.Sprintf("Error initializing client: %s", err))
+			return 1
+		}
+		qo := &api.QueryOptions{
+			Namespace: c.Meta.namespace,
+		}
+
+		// Get and list all variables at nomad/job-templates
+		vars, _, err := client.Variables().PrefixList("nomad/job-templates/", qo)
+		if err != nil {
+			c.Ui.Error(fmt.Sprintf("Error retrieving vars: %s", err))
+			return 1
+		}
+
+		if len(vars) == 0 {
+			c.Ui.Error("No variables in nomad/job-templates")
+			return 1
+		} else {
+			c.Ui.Output("Use nomad job init -template=<template> with any of the following:")
+			for _, v := range vars {
+				fmt.Println(strings.TrimPrefix(v.Path, "nomad/job-templates/"))
+			}
+		}
+		return 0
+	} else if template != "" {
+		// Get the HTTP client
+		client, err := c.Meta.Client()
+		if err != nil {
+			c.Ui.Error(fmt.Sprintf("Error initializing: %s", err))
+			return 1
+		}
+
+		qo := &api.QueryOptions{
+			Namespace: c.Meta.namespace,
+		}
+		sv, _, err := client.Variables().Read("nomad/job-templates/"+template, qo)
+		if err != nil {
+			if err.Error() == "variable not found" {
+				c.Ui.Warn(errVariableNotFound)
+				return 1
+			}
+			c.Ui.Error(fmt.Sprintf("Error retrieving variable: %s", err))
+			return 1
+		}
+		// If the user provided an item key, return that value instead of the whole
+		// object
+		if v, ok := sv.Items["template"]; ok {
+			c.Ui.Output(fmt.Sprintf("Initializing a job template from %s", template))
+			jobSpec = []byte(v)
+		} else {
+			c.Ui.Error(fmt.Sprintf("Variable does not contain %q item", args[1]))
```

Comment:
This error message suggests we're looking for the command argument but we're actually looking for the field `template`, right? I think we want to tell the user that the variable that stores the template is malformed and missing a `template` field. Maybe even refer them to the UI page with the editor?

Reply:
Good call, I hastily copied this from our more open-ended get variable code.